snappi_ixnetwork/trafficitem.py

```diff
@@ -182,18 +284,311 @@ class TrafficItem(CustomField):
+            "arg1": href,
+            "arg2": ["/traffic/trafficItem/descendant-or-self::*"],
+            "arg3": True,
+            "arg4": "json",
+        }
+        res = self._api._request("POST", url=url, payload=payload)
+        return json.loads(res["result"])
+
+    def _importconfig(self, imports):
+        imports["xpath"] = "/"
+        href = "%sresourceManager" % self._api._ixnetwork.href
+        url = "%s/operations/importconfig" % href
+        import json
+
+        payload = {
+            "arg1": href,
+            "arg2": json.dumps(imports),
+            "arg3": False,
+            "arg4": "suppressNothing",
+            "arg5": True,
+        }
+        try:
+            # TODO for larger config rest api is throwing error,
+            # with no url found, when the first response is 202 (in-progress)
+            # its keep checking the status of the url with 1 sec sleep, and
+            # after a while error is thrown. but could see the configuration
+            # applied at Ixnetwork. (Need to check with Eng team)
+            response = self._api._request("POST", url=url, payload=payload)
+        except Exception:
+            return
+        if (
+            response["result"].get("errata") is not None
+            and response["result"]["errata"] != []
+        ):
+            self._api.get_json_import_errors()
+            raise SnappiIxnException(
+                400, "{}".format(response["result"]["errata"])
+            )
+
+    def get_ports_encap(self, config):
+        ixn = self._api.assistant._ixnetwork
+        myfilter = [{"property": "name", "regex": ".*"}]
+        url, payload = self._get_search_payload(
+            "/",
+            "(?i)^(vport)$",
+            ["name"],
+            myfilter,
+        )
+        result = ixn._connection._execute(url, payload)
+        vports = {}
+        for vp in result:
+            if vp.get("vport") is None:
+                continue
+            for v in vp["vport"]:
+                vports[v["name"]] = v["xpath"]
+        return vports
+
+    def get_device_encap(self, config):
+        if len(config.devices) == 0:
+            return {}
+        dev_names = []
+        for f in config.flows:
+            if f.tx_rx.choice == "port":
+                continue
+            dev_names.extend(f.tx_rx.device.tx_names)
+            dev_names.extend(f.tx_rx.device.rx_names)
+        dev_names = list(set(dev_names))
+        if dev_names == []:
+            return {}
+        selects = []
+        url = None
+        for name in dev_names:
+            href = self._api.get_ixn_href(name)
+            if href is None:
+                # TODO change to more detailed message
+                raise Exception(
+                    "href is not found for device name {}".format(name)
+                )
+            href = href.split("ixnetwork")[-1]
+            query = "^(xpath|ethernet|ipv4|ipv6)$"
+            if "ipv4" in href or "ipv6" in href:
```

Comment:
We already stored that info in self._api.._device_encap. So we will get the type using self._api.get_device_encap.
Otherwise we need to use more regex to extract actual type.

Reply:
is the device type stored in current main branch or was it added in device compaction?

Reply:
This was added log back probably day 1. I copy (self._api.get_device_encap) that from your brunch.